src/main/java/org/spongepowered/api/entity/player/tab/TabList.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.player.tab;
+
+import org.spongepowered.api.text.message.Message;
+
+import com.google.common.base.Optional;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface TabList {
+
+    /**
+     * Sets this list's header.
+     * 
+     * @param header The new header
+     */
+    public void setHeader(Message header);
+
+    /**
+     * Gets this list's header.
+     * 
+     * @return The current header
+     */
+    public Message getHeader();
+
+    /**
+     * Sets this list's footer.
+     * 
+     * @param footer The new footer
+     */
+    public void setFooter(Message footer);
+
+    /**
+     * Gets this list's footer.
+     * 
+     * @return The current footer
+     */
+    public Message getFooter();
+
+    /**
+     * Gets the players on the list.
+     * 
+     * @return The players on the list
+     */
+    public List<PlayerTabInfo> getPlayers();
+
+    /**
+     * Adds a player to the list.
+     * 
+     * @throws IllegalArgumentException when it attempts to add a player already
+     *         on the list. This is to prevent modification of a
+     *         {@link PlayerTabInfo} by overwriting it
+     * 
+     * @param player The player to add
+     */
+    public void addPlayer(PlayerTabInfo player) throws IllegalArgumentException;
```

Comment:
Isn't this already possible through the list returned by the method above? Or is the list immutable?